node/src/Pos/Block/Worker.hs

```diff
@@ -83,14 +83,19 @@ blkWorkers =
   where
     merge = mconcatPair . map (first pure)
 
--- Action which should be done when new slot starts.
+-- FIXME [CSL-1576] Metric worker should be independent of block creator.
+-- TODO [CSL-1606] Using 'fork' here is quite bad, it's a temporary solution.
 blkOnNewSlot :: WorkMode ssc ctx m => (WorkerSpec m, OutSpecs)
 blkOnNewSlot =
     onNewSlotWorker True announceBlockOuts $ \slotId sendActions ->
         recoveryCommGuard $
         () <$
         metricWorker slotId `concurrently`
-        blockCreator slotId sendActions
+        void
+            (fork $
```

Comment:
You added a fork in one place and added it in another?

Reply:
Yes, but only for this particular worker. For all other `onNewSlot` there is no `fork` because it's not needed. I would like to get rid of this one as well, but in another issue.
See this commit's message: https://github.com/input-output-hk/cardano-sl/commit/7f426d0bdbfdfdb8be92620950b614a85c6339ad
Also see CSL-1606 for broader explanation of `trickier`.